main.go

```diff
@@ -65,6 +70,7 @@ func main() {
 	ctx := signals.SetupSignalHandler()
 	logger := klog.FromContext(ctx)
 
+	server := startHealthProbeServer(healthProbeAddr, logger)
```

Comment:
this always return OK, so I expect this to run after the controllers are running ... not the best thing but seems better aproximation

Reply:
you are absolutely right, but in this case the non-leader replicas will not be ready until they get the leadership because of the readinessProbe definition, I guess. So I'm not sure how we should handle this, I'll take a look at other SIG/k8s projects that use leader election to find a possible workaround for this.

Reply:
let's leave it this way, better safe than trying to overoptimize